File: src/01-easy/pirates-treasure/solution.vb
Module Solution
    Sub Main()
        Dim w=CInt(Console.ReadLine()),h=CInt(Console.ReadLine())
        Dim g(h-1,w-1) As Integer
        For y=0 To h-1:Dim r=Console.ReadLine().Split(" "c):For x=0 To w-1:g(y,x)=CInt(r(x)):Next:Next
        Dim dx={-1,-1,-1,0,0,1,1,1},dy={-1,0,1,-1,1,-1,0,1}
        For y=0 To h-1:For x=0 To w-1:If g(y,x)=0 Then
            Dim ok=True
            For i=0 To 7:Dim nx=x+dx(i),ny=y+dy(i)
                If nx>=0 AndAlso nx<w AndAlso ny>=0 AndAlso ny<h AndAlso g(ny,nx)<>1 Then ok=False
            Next
            If ok Then Console.WriteLine(x & " " & y):End
        End If:Next:Next
    End Sub
End Module

File: src/01-easy/pirates-treasure/solution.cs
using System;
class Solution{
    static void Main(){
        int w=int.Parse(Console.ReadLine()),h=int.Parse(Console.ReadLine());
        var g=new int[h,w];
        for(int y=0;y<h;y++){var r=Console.ReadLine().Split(' ');for(int x=0;x<w;x++)g[y,x]=int.Parse(r[x]);}
        int[] dx={-1,-1,-1,0,0,1,1,1},dy={-1,0,1,-1,1,-1,0,1};
        for(int y=0;y<h;y++)for(int x=0;x<w;x++)if(g[y,x]==0){
            bool ok=true;
            for(int i=0;i<8;i++){int nx=x+dx[i],ny=y+dy[i];if(nx>=0&&nx<w&&ny>=0&&ny<h&&g[ny,nx]!=1)ok=false;}
            if(ok){Console.WriteLine(x+" "+y);return;}
        }
    }
}
